### PR TITLE
Add [credential record/authenticatorDisplayName] handling to RP operations

### LINKED ISSUE #2156: Add `authenticatorDisplayName` to Step 27 and Step 23 of the registration and authentication ceremonies respectively
#1880 added the optional item `authenticatorDisplayName` to [_Credential Record_](https://w3c.github.io/webauthn/#abstract-opdef-credential-record-authenticatordisplayname); however it was not added to Step 27 of the registration ceremony nor Step 23 of the authentication ceremony. Seeing how `attestationObject` and `attestationClientDataJSON` exist in Step 27, it would be nice to also add `authenticatorDisplayName`. Honestly, I was under the impression `authenticatorDisplayName` was only a `credProps` thing; had it existed in the ceremony criteria section for data to store, I don't think I would have made that mistake.

Something like below would be nice:

> The new [credential record](https://w3c.github.io/webauthn/#credential-record) MAY also include the following OPTIONAL contents:
>
> [`attestationObject`](https://w3c.github.io/webauthn/#abstract-opdef-credential-record-attestationobject)
>
>  _`response.`_[`attestationObject`](https://w3c.github.io/webauthn/#dom-authenticatorattestationresponse-attestationobject).
>
> [`attestationClientDataJSON`](https://w3c.github.io/webauthn/#abstract-opdef-credential-record-attestationclientdatajson)
>
>  _`response.`_[`clientDataJSON`](https://w3c.github.io/webauthn/#dom-authenticatorresponse-clientdatajson).
>
> [`authenticatorDisplayName`](https://w3c.github.io/webauthn/#abstract-opdef-credential-record-authenticatordisplayname)
>
>  _`clientExtensionResults.credProps.`_[`authenticatorDisplayName`](https://w3c.github.io/webauthn/#dom-credentialpropertiesoutput-authenticatordisplayname) or RP-provided value.

### PR DESCRIPTION
**Pull request by emlun**
Fixes #2156.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/2163.html" title="Last updated on Sep 30, 2024, 1:58 PM UTC (8b13724)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/2163/efdf948...8b13724.html" title="Last updated on Sep 30, 2024, 1:58 PM UTC (8b13724)">Diff</a>

### CODE REVIEW

**Review by timcappalli — APPROVED** [review]

**Review by sbweeden — APPROVED** [review]
LGTM

**Review by selfissued — APPROVED** [review]
Approved during the 9-Oct-24 call

**Comment by MasterKale** [comment]
From the WG call today: This is fine to merge any time once the merge conflicts are resolved.

**Pull request merged**
